### PR TITLE
[No Ticket] - String Operations - Replace

### PR DESCRIPTION
**Pull request by jmcadams-r7**
## Proposed Changes
This PR adds the Replace function to the string Plugin

## Testing: 
```
RMT-MBP-5357:string jmcadams$ ./run.sh -R tests/replace.json -j
Running: cat tests/replace.json | docker run --rm   -i rapid7/string:1.4.0  run | grep -- ^\{ | jq -r '.body | try(.log | split("\n") | .[]),.output'
rapid7/String Operations:1.3.1. Step name: replace
in_string: {"id":"R1155","title":"Account Executive, Commercial Accounts","location":"US - MA - Boston"}
{"id":"R2226","title":"Account Executive, Commercial Accounts","location":"US - TX - Austin"}
find_string: id
find_string: ID
rapid7/String Operations:1.3.1. Step name: replace
in_string: {"id":"R1155","title":"Account Executive, Commercial Accounts","location":"US - MA - Boston"}
{"id":"R2226","title":"Account Executive, Commercial Accounts","location":"US - TX - Austin"}
find_string: id
find_string: ID

{
  "result_string": "{\"ID\":\"R1155\",\"title\":\"Account Executive, Commercial Accounts\",\"location\":\"US - MA - Boston\"}\n{\"ID\":\"R2226\",\"title\":\"Account Executive, Commercial Accounts\",\"location\":\"US - TX - Austin\"}"
}
```

## Workflow: 
image

### CODE REVIEW

**Review by mrinehart-r7 — APPROVED** [review]

File: plugins/string/plugin.spec.yaml [hunk 1]
```diff
@@ -172,3 +172,31 @@ actions:
...
+        description: The string that will replace the parts that are found. If left blank the characters to find will be deleted
+        type: string
+        required: false
+        example: "replacement"
+    output:
+      result_string:
+        title: Result String
+        description: The string after replacement
+        type: string
+        required: false
```
Comment (mrinehart-r7) [inline plugins/string/plugin.spec.yaml]:
```suggestion
        required: true
```
Check to make sure an empty string validates fine in product

File: plugins/string/komand_string/actions/replace/action.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        super(self.__class__, self).__init__(
+            name="replace", description=Component.DESCRIPTION, input=ReplaceInput(), output=ReplaceOutput()
+        )
+
+    def run(self, params={}):
+        in_string = params.get(Input.IN_STRING)
+        find_string = params.get(Input.STRING_PART_TO_FIND)
+        replace_string = params.get(Input.REPLACEMENT_VALUE)
+        if not replace_string:
+            replace_string = ""
```
Comment (mrinehart-r7) [inline plugins/string/komand_string/actions/replace/action.py]:
```suggestion
        replace_string = params.get(Input.REPLACEMENT_VALUE, "")
```

File: plugins/string/komand_string/actions/replace/action.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+        self.logger.info(f"in_string: {in_string}")
+        self.logger.info(f"find_string: {find_string}")
+        self.logger.info(f"find_string: {replace_string}")
```
Comment (mrinehart-r7) [inline plugins/string/komand_string/actions/replace/action.py]:
```suggestion
        self.logger.info(f"replace_string: {replace_string}")
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Review by mrinehart-r7 — DISMISSED** [review]

**Review by mrinehart-r7 — APPROVED** [review]

**Review by aholland-r7 — APPROVED** [review]

**Review by pmara-r7 — APPROVED** [review]
reviewed as a team

**Pull request merged**